Networking/Networking/Model/MetaData.swift

```diff
@@ -39,18 +47,92 @@ private extension MetaData {
     }
 }
 
-private extension AnyCodable {
-    func getAsString() -> String {
-        switch value {
-        case let string as String:
-            return string
-        default:
-            if JSONSerialization.isValidJSONObject(value) {
-                let data = (try? JSONSerialization.data(withJSONObject: value)) ?? Data()
-                return String(data: data, encoding: .utf8) ?? ""
-            } else {
-                return "\(value)"
+public struct MetaDataValue: Codable, Equatable, Sendable {
+    private let valueHolder: ValueHolder
+
+    public var stringValue: String {
+        valueHolder.stringValue
+    }
+    public var rawValue: String {
+        valueHolder.rawValue
+    }
+
+    public var isJson: Bool {
+        switch valueHolder {
+        case .string:
+            return false
+        case .json:
+            return true
+        }
+    }
+
+    public init(rawValue: String) {
+        let data = Data(rawValue.utf8)
+        if let jsonObject = try? JSONDecoder().decode(AnyCodable.self, from: data) {
+            valueHolder = try! ValueHolder(from: jsonObject.value)
+        } else {
+            valueHolder = .string(rawValue.removingPrefix("\"").removingSuffix("\""))
+        }
+    }
+
+    public init(from decoder: Decoder) throws {
+        let codable = try decoder.singleValueContainer().decode(AnyCodable.self)
+        valueHolder = try ValueHolder(from: codable.value)
+    }
+
+    public func encode(to encoder: Encoder) throws {
+        switch valueHolder {
+        case .string(let value):
+            try value.encode(to: encoder)
+        case .json(let json):
+            let encodable = try JSONDecoder().decode(AnyCodable.self, from: Data(json.utf8))
+            try encodable.encode(to: encoder)
+        }
+    }
+}
+
+private extension MetaDataValue {
+    private enum ValueHolder: Codable, Equatable {
+        case string(_ value: String)
+        case json(_ json: String)
```

Comment:
I played with a different approach here, instead of making the backing data a `String`, we can have it as `Codable` directly, I don't have a strong opinion on which approach is best, so I kept the `String` based implementation.

```swift
public struct MetaDataValue: Codable, Equatable, Sendable {
    private let valueHolder: ValueHolder

    public var stringValue: String {
        valueHolder.stringValue
    }
    public var rawValue: String {
        valueHolder.rawValue
    }

    public var isJson: Bool {
        switch valueHolder {
        case .string:
            return false
        case .json:
            return true
        }
    }

    public init(rawValue: String) {
        let data = Data(rawValue.utf8)
        if let jsonObject = try? JSONDecoder().decode(AnyCodable.self, from: data) {
            valueHolder = try! ValueHolder(from: jsonObject.value)
        } else {
            valueHolder = .string(rawValue.removingPrefix("\"").removingSuffix("\""))
        }
    }

    public init(from decoder: Decoder) throws {
        valueHolder = try ValueHolder(from: decoder)
    }

    public func encode(to encoder: Encoder) throws {
        try valueHolder.encode(to: encoder)
    }
}

private extension MetaDataValue {
    private enum ValueHolder: Codable, Equatable {
        case string(_ value: String)
        case json(_ json: Codable)

        var stringValue: String {
            switch self {
            case .string(let value):
                return value
            case .json(let json):
                return json.asJSONString()
            }
        }

        var rawValue: String {
            switch self {
            case .string(let value):
                return "\"\(value)\""
            case .json(let json):
                return json.asJSONString()
            }
        }

        init(from decoder: any Decoder) throws {
            let container = try decoder.singleValueContainer()
            let value = try container.decode(AnyCodable.self)
            try self.init(from: value.value)
        }

        init(from object: Any) throws {
            switch object {
            case let value as String:
                self = .string(value)
            case is Bool:
                self = .string(String(describing: object))
            case is any Numeric:
                self = .string(String(describing: object))
            default:
                self = .json(AnyCodable(object))
            }
        }

        func encode(to encoder: Encoder) throws {
            switch self {
            case .string(let value):
                try value.encode(to: encoder)
            case .json(let json):
                try json.encode(to: encoder)
            }
        }

        static func == (lhs: ValueHolder, rhs: ValueHolder) -> Bool {
            return lhs.stringValue == rhs.stringValue
        }
    }
}

private extension Encodable {
    func asJSONString() -> String {
        if let data = try? JSONEncoder().encode(self) {
            return String(data: data, encoding: .utf8) ?? ""
        }
        return ""
    }
}
```

Reply:
Interesting alternative approach with `Codable`. My hunch here is that for the current use case where we only need to validate if a value is valid JSON and display it in a read-only field, the String approach is more stragithforward.